src/rust/generators/osquery-subgraph-generator/src/parsers/mod.rs

```diff
@@ -1,108 +1,36 @@
-use std::{
-    collections::HashMap,
-    convert::TryFrom,
-};
-
 use grapl_graph_descriptions::graph_description::*;
-use regex::Regex;
 use serde::{
-    de::DeserializeOwned,
     Deserialize,
     Serialize,
 };
-use serde_json::Value;
 
 mod grapl_pack;
 
-#[derive(Serialize, Deserialize)]
-struct OSQueryResponse<C> {
-    #[serde(rename = "hostIdentifier")]
-    host_identifier: String,
-    #[serde(rename = "calendarTime")]
-    calendar_time: String,
-    #[serde(rename = "unixTime")]
-    unix_time: u64,
-    columns: C,
-    action: OSQueryAction,
+#[derive(Serialize, Deserialize, Clone, Hash)]
+#[serde(tag = "name")]
+pub enum OSQueryEvent {
+    #[serde(rename = "pack_grapl_processes")]
+    Process(grapl_pack::processes::ProcessEvent),
+    #[serde(rename = "pack_grapl_process-files")]
+    ProcessFileAction(grapl_pack::process_files::ProcessFileInteractionEvent),
+    #[serde(rename = "pack_grapl_files")]
+    File(grapl_pack::files::FileEvent),
 }
 
-#[derive(Serialize, Deserialize)]
-enum OSQueryAction {
-    #[serde(rename = "added")]
+#[derive(Serialize, Deserialize, Debug, Clone, Hash)]
+#[serde(rename_all = "camelCase")]
+pub(self) enum OSQueryAction {
```

Comment:
TIL pub(self)
https://doc.rust-lang.org/reference/visibility-and-privacy.html

And apparently there are even more - I had no idea there were more than pub(crate).